portfolio/src/main/java/com/google/sps/servlets/RegionDataServlet.java

```diff
@@ -29,44 +31,45 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   }
 
   @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+  public synchronized void doPost(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
     String region = request.getParameter("region");
 
-    Entity voteEntity = new Entity("Vote");
-    voteEntity.setProperty("name", region);
+    Entity regionEntity = new Entity("Region", region);
+    regionEntity.setProperty("name", region);
 
-    writeToDatabase(voteEntity);
+    Map<String, Long> currentVotes = getVotes();
```

Comment:
This is better; You now only keep one entity per region, so your storage is O(countries) rather than O(votes).
You are, however, still reading the entire vote map in order to change a single entity.
Could you do datastore.get(regionEntity) here (in the post method) to retrieve only the entity you need?

Reply:
I fixed it so that it only retrieves the necessary entity for doPost(). Now the only time it makes the entire map is in the doGet() method. Thanks for the feedback!